source/common/stream_info/stream_info_impl.h

```diff
@@ -183,6 +183,14 @@ struct StreamInfoImpl : public StreamInfo {
     return downstream_remote_address_;
   }
 
+  void setDownstreamSslConnection(const Ssl::ConnectionInfo* connection_info) override {
+    downstream_ssl_info_ = connection_info;
```

Comment:
I'm slightly worried that StreamInfoImpl might outlive Ssl::ConnectionInfo, will that be the case?

Reply:
From what I can tell it seems like `Ssl::ConnectionInfo/Network::Connection` comes from the `FilterManagerImpl` which should outlive the Filters that the StreamInfoImpl belongs to. Do you think it would be useful to expose the ConnectionInfo as a shared ptr to avoid potential lifetime issues in the future?

Reply:
No I think it is fine to leave as is.